operational_analysis/methods/plant_analysis.py

```diff
@@ -854,8 +860,33 @@ def sample_long_term_reanalysis(self, n, r):
             wd_aggregate = np.pi-np.arctan2(-long_term_temp[r + '_u_ms'],long_term_temp[r + '_v_ms']) # Calculate wind direction
             long_term_reg_inputs = pd.concat([long_term_reg_inputs, np.sin(wd_aggregate), np.cos(wd_aggregate)], axis=1)
         
+        # Store result in dictionary
+        self.long_term_sampling[(r, n)] = long_term_reg_inputs
+        self.long_term_sampling_sd[(r, n)] = ws_aggregate_sd
+        
         # Return result            
-        return long_term_reg_inputs
+        return [long_term_reg_inputs,ws_aggregate_sd]
+
+    @logged_method_call
+    def apply_IAV(self, long_term_reg_inputs, ws_aggregate_sd_IAV, r):
+        """
+        This function calculates IAV for the long-term wind speed data
+        
+        Args:
+           long_term_reg_inputs(:obj:`pandas.DataFrame`): Long-term regression inputs
+           ws_aggregate_sd(:obj:`pandas.DataFrame`): Long-term wind speed stdev, from which IAV will be derived
+           r(:obj:`string`): The reanalysis product used for the current Monte Carlo iteration
+        Returns:
+           :obj:`pandas.DataFrame`: the IAV-sampled monthly/daily regression inputs
+        """
+    
+        if self.time_resolution == 'D':
+            ws_aggregate_sd_IAV = np.repeat(ws_aggregate_sd_IAV, np.ceil(self.num_days_lt).astype(int))
+        ws_aggregate_sd_IAV.reset_index(drop=True,inplace=True) # TODO: This rectifies a 0-indexing issue. Needs refactoring.
+        iav = np.random.normal(1, ws_aggregate_sd_IAV/long_term_reg_inputs[r])        
+        long_term_reg_inputs_IAV = long_term_reg_inputs.copy()
+        long_term_reg_inputs_IAV[r] = long_term_reg_inputs_IAV[r] * iav
```

Comment:
It looks like you're only updating one of the components of long_term_reg_inputs_IAV[r], but returning the whole variable (which has possibly been re-indexed and re-sampled in previous lines)?

Reply:
Yes, only the wind speed component of long_term_reg_inputs_IAV is updated (since no IAV is applied to wind direction and temperature, when these are present in long_term_reg_inputs_IAV). Not sure I understood the issue here..

Reply:
Ok. I wonder if there is a way to get rid of the "r" parameter? Didn't we refactor this at some point to self._mc_params? What happened to that?

Reply:
We had created a self._rean_vars variable to save whether temperature and wind components (i.e. wind direction) are chosen by the user. We didn't include wind speed itself in there - probably because there's no option to not use wind speed. 
So for example sample_long_term_reanalysis has a first chunk for wind speed, and then a second chunk for temperature and wind direction (when they are used), controlled by:
namescol = [r + '_' + var for var in self._rean_vars]

Reply:
Here - we refactored it to the self._run variable in the turbine_lon_term_gross_energy PR:

https://github.com/NREL/OpenOA/blob/2786cc725e797ef5275083801fee154d0ccf0489/operational_analysis/methods/turbine_long_term_gross_energy.py#L158

And then you can get "r" through self._run.reanalysis_product

https://github.com/NREL/OpenOA/blob/2786cc725e797ef5275083801fee154d0ccf0489/operational_analysis/methods/turbine_long_term_gross_energy.py#L318

Reply:
Yes, that's how both `long_term_reg_inputs_IAV` and `apply_IAV` are called:

```
            # Get long-term regression inputs
            [reg_inputs_lt_noIAV, reg_lt_stdev] = self.sample_long_term_reanalysis(self._run.num_years_windiness,
                                                             self._run.reanalysis_product)

            # Apply IAV to long-term regression inputs
            reg_inputs_lt = self.apply_IAV(reg_inputs_lt_noIAV, reg_lt_stdev, self._run.reanalysis_product)
            
```

Am I missing something?

Reply:
There is no need to pass in "self._" anything to these methods because you already have access to "self" automatically. 

Can't we just use self._run.reanalysis_product self. _run.num_years_windiness from within self.sample_long_term_reanalysis and self.apply_IAV, getting rid of both arguments to sample_long_term_reanalysis and and getting rid of the 3rd argument to apply_IAV?

Reply:
Ahh, got it! I've now changed this.